### PR TITLE
Correct issue with new phone number deletion logic

### PR DESCRIPTION
**Pull request by callumknights**
NilClass errors where appearing for certain WWorgs, this should account for that.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/controllers/admin/worldwide_offices_controller.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def update
 
     if (contact_numbers_attributes = update_params.dig(:contact_attributes, :contact_numbers_attributes))
       contact_numbers_attributes.each do |_id, contact_number_attribute|
-        contact_number_attribute[:_destroy] = true if contact_number_attribute[:label].empty? && contact_number_attribute[:number].empty?
+        contact_number_attribute[:_destroy] = true unless contact_number_attribute[:label].present? && contact_number_attribute[:number].present?
```
Comment (yndajas) [inline app/controllers/admin/worldwide_offices_controller.rb:26]: Should the `&&` be `||` here now? Or is the contact number only valid if **both** fields are present?

File: app/controllers/admin/worldwide_offices_controller.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def update
 
     if (contact_numbers_attributes = update_params.dig(:contact_attributes, :contact_numbers_attributes))
       contact_numbers_attributes.each do |_id, contact_number_attribute|
```
Comment (yndajas) [inline app/controllers/admin/worldwide_offices_controller.rb:25]: Could also do this in a separate commit to simplify this code, but it's not important!

https://github.com/alphagov/whitehall/pull/9084/files#r1624701069

File: app/controllers/admin/worldwide_offices_controller.rb (hunk 1 above)
Comment (brucebolt) [inline app/controllers/admin/worldwide_offices_controller.rb:26]: Will this mark for destruction if any one of either the number and label are not present?

Two tests with one for each of `label` and `number` being blank would clarify this.

**Review by brucebolt — COMMENTED** [review]
Could you update the `PUT :update deletes contact numbers that are marked as to be destroyed` test to reflect that we've realised `label` and `number` aren't present when a contact has been deleted in the UI.  That will let us catch if this issue is reintroduced in the future.

File: app/controllers/admin/worldwide_offices_controller.rb (hunk 1 above)
Comment (callumknights) [reply app/controllers/admin/worldwide_offices_controller.rb:26]: Only if both are present that's correct

Comment (callumknights) [reply app/controllers/admin/worldwide_offices_controller.rb:26]: Updated, can confirm it will mark a contact number for destruction if it's missing either field.

File: test/functional/admin/worldwide_offices_controller_test.rb [hunk 3]
```diff
@@ -292,8 +292,8 @@ class Admin::WorldwideOfficesControllerTest < ActionController::TestCase
               contact_numbers_attributes: {
                 "0" => {
                   id: contact_number.id,
-                  label: contact_number.label,
-                  number: contact_number.number,
+                  label: "",
+                  number: "",
```
Comment (brucebolt) [inline test/functional/admin/worldwide_offices_controller_test.rb:296]: It appears these values aren't being included at all in the payload, hence the `undefined method `empty?' for nil:NilClass` error.  So both these lines can just be deleted.

File: test/functional/admin/worldwide_offices_controller_test.rb [hunk 4]
```diff
@@ -306,6 +306,58 @@ class Admin::WorldwideOfficesControllerTest < ActionController::TestCase
...
+              },
+            },
+          },
+          id: office,
+          worldwide_organisation_id: worldwide_organisation,
+        }
+
+    assert_not ContactNumber.exists?(contact_number.id)
+  end
+
```
Comment (brucebolt) [inline test/functional/admin/worldwide_offices_controller_test.rb:360]: Thanks for adding both these tests, they look great.

File: test/functional/admin/worldwide_offices_controller_test.rb (hunk 3 above)
Comment (callumknights) [reply test/functional/admin/worldwide_offices_controller_test.rb]: Sorted :)

File: test/functional/admin/worldwide_offices_controller_test.rb [hunk 5]
```diff
@@ -306,6 +304,58 @@ class Admin::WorldwideOfficesControllerTest < ActionController::TestCase
...
+                  number: "",
+                },
+              },
+            },
+          },
+          id: office,
+          worldwide_organisation_id: worldwide_organisation,
+        }
+
+    assert_not ContactNumber.exists?(contact_number.id)
```
Comment (brucebolt) [inline test/functional/admin/worldwide_offices_controller_test.rb:356]: We shouldn't be deleting the record if a user leaves one field blank (as they will lose the data they've entered into the other field).  Instead there should be a validation error displayed (so change the `&&` to `||` in the controller, or switch back to using `if` but with `blank?` instead of `empty?`, as `nil` responds to that method).

Comment (callumknights) [reply test/functional/admin/worldwide_offices_controller_test.rb]: This should all be solved now

**Review by brucebolt — APPROVED** [review]
